### ISSUE TITLE
single-table inheritance is not working

### ISSUE THREAD

**Issue by hmarcelodn** [typeorm/typeorm#596]
Hello,

I have created 2 very simple tables using single-table relationship. Tables are created but child entity columns are not.

Check the code I used:

**Assesment Parent Class:**

```ts
import { Entity, Column, TableInheritance, DiscriminatorColumn, PrimaryColumn } from "typeorm";

@Entity("assesment")
@TableInheritance("single-table")
@DiscriminatorColumn({ name: "type", type: "varchar" })
export class Assesment{

    @PrimaryColumn("int")
    id: number;
   
}
```

**Child Class**

```ts
import { ClassEntityChild, Column, OneToMany, DiscriminatorValue, SingleEntityChild } from "typeorm";
import { Assesment } from "./assesment";

@SingleEntityChild()
@DiscriminatorValue(1)
export class AssesmentQuiz extends Assesment {    

    constructor(){
        super();
    }

    @Column()
    test: string;

}
```

Assesment columns were created and AssesmentQuiz were not. I dont see the test column.

Any ideas?

Thanks.

**Comment by ryuuji3** [comment]
In a similar scenario, I tried the same as above for demonstration purposes without the `@DiscriminatorValue()` I still had the problem described.

The following: 


```javascript
// Employee.js
@Entity()
@TableInheritance('single-table')
@DiscriminatorColumn({ name: 'employeeType', type: 'int'})
export default class Employee {

  @PrimaryGeneratedColumn()
  id: number

  @Column()
  firstName: string

  @Column()
  lastName: string

  @Column()
  birthday: Date

  @Column('text')
  address: AddressI

  @Column('text')
  contactInfo: PhoneNumberI[]

}

// Manager.js

@SingleEntityChild()
export default class Manager extends Employee {
  @Column()
  test: string
}
```

yields:

```sql
MariaDB [schedules]> describe employee;
+--------------+--------------+------+-----+---------+----------------+
| Field        | Type         | Null | Key | Default | Extra          |
+--------------+--------------+------+-----+---------+----------------+
| id           | int(11)      | NO   | PRI | NULL    | auto_increment |
| firstName    | varchar(255) | NO   |     | NULL    |                |
| lastName     | varchar(255) | NO   |     | NULL    |                |
| birthday     | datetime     | NO   |     | NULL    |                |
| address      | text         | NO   |     | NULL    |                |
| contactInfo  | text         | NO   |     | NULL    |                |
| employeeType | int(11)      | NO   | MUL | NULL    |                |
+--------------+--------------+------+-----+---------+----------------+
```

**Comment by pleerock** [comment]
This problem must be fixed in `0.2.0-alpha.11`, please check and let me know if its not working for still

**Comment by MohsinGhani** [comment]
Hello @pleerock, I did single-table inheritance and it's working fine but is there any way to do It without an additional discriminator column?
